src/main/java/tool/feedback/Feedback.java

```diff
@@ -0,0 +1,142 @@
+package tool.feedback;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import tool.designpatterns.DesignPattern;
+import tool.designpatterns.Pattern;
+
+/**
+ * Simple class to store and manage feedback to the user.
+ */
+@DesignPattern(pattern = {Pattern.IMMUTABLE})
+public final class Feedback {
+
+    private boolean isError;
+    private final String message;
+    private final List<Feedback> children;
+    private final FeedbackTrace stackTrace;
+
+    private static final String LINE_PREFIX = "  ";
+
+    private Feedback(
+        boolean isError, String message, FeedbackTrace stackTrace, List<Feedback> children) {
+        this.isError = isError;
+        this.message = message;
+        this.stackTrace = stackTrace;
+        this.children = new ArrayList<>();
+
+        if (children != null) {
+            for (Feedback child : children) {
+                if (child.getIsError()) {
+                    this.isError = true;
+                    this.children.add(child);
+                }
+            }
+        }
+    }
+
+    /**
+     * Get a new feedback that is an error with the given message and stacktrace. It does not have
+     * any children and is therefore the leaf feedback.
+     *
+     * @param message    what caused the error?
+     * @param stackTrace the element where the error occured.
+     *
+     * @return the new feedback.
+     */
+    public static Feedback getNoChildFeedback(
+        String message, FeedbackTrace stackTrace) {
+        return new Feedback(true, message, stackTrace, null);
+    }
+
+    /**
+     * Get a new feedback that has children, in case any of the children has any errors this will
```

Comment:
I don't understand the comment in combination with the code. Do errors relate to isError that is set to false? What does it mean that it is successfull?

Reply:
It being an error means that the "isError" variable is true, it also means that one of the leaf nodes in the "feedback-tree" has an error and so should also have an error message.